server/src/test/java/org/opensearch/index/shard/RemoteStoreRefreshListenerTests.java

```diff
@@ -217,7 +217,8 @@ public void testRefreshSuccessOnFirstAttempt() throws Exception {
         // We spy on IndexShard.getReplicationTracker() to validate that we have tried running remote time as per the expectation.
         CountDownLatch refreshCountLatch = new CountDownLatch(succeedOnAttempt);
         // We spy on IndexShard.getEngine() to validate that we have successfully hit the terminal code for ascertaining successful upload.
-        CountDownLatch successLatch = new CountDownLatch(2);
+        // Value has been set as 3 as during a successful upload IndexShard.getEngine() is hit thrice and with mockito we are counting down
+        CountDownLatch successLatch = new CountDownLatch(3);
```

Comment:
thanks . won't we need to do same for other tests as well ?

Reply:
added the missed condition..